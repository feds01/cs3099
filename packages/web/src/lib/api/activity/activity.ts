/**
 * Generated by orval v6.6.4 üç∫
 * Do not edit manually.
 * Iamus API
 * This is a REST API for interfacing with Iamus. This API provides endpoints for interacting with user information, submissions, and reviews.
 * OpenAPI spec version: 1.0.0
 */
import type {
    GetUserUsernameFeed200,
    ApiErrorResponse,
    GetUserUsernameFeedParams,
    GetActivityId200,
    GetActivity200,
    GetActivityParams,
} from '.././models';
import { customInstance } from '.././mutator/custom-instance';
import { useQuery, UseQueryOptions, QueryFunction, UseQueryResult, QueryKey } from 'react-query';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
type AsyncReturnType<T extends (...args: any) => Promise<any>> = T extends (...args: any) => Promise<infer R> ? R : any;

/**
 * Get a paginated list of activities for a particular user=
 * @summary List activities for a given user
 */
export const getUserUsernameFeed = (username: string, params?: GetUserUsernameFeedParams) => {
    return customInstance<GetUserUsernameFeed200>({ url: `/user/${username}/feed`, method: 'get', params });
};

export const getGetUserUsernameFeedQueryKey = (username: string, params?: GetUserUsernameFeedParams) => [
    `/user/${username}/feed`,
    ...(params ? [params] : []),
];

export const useGetUserUsernameFeed = <TData = AsyncReturnType<typeof getUserUsernameFeed>, TError = ApiErrorResponse>(
    username: string,
    params?: GetUserUsernameFeedParams,
    options?: { query?: UseQueryOptions<AsyncReturnType<typeof getUserUsernameFeed>, TError, TData> },
): UseQueryResult<TData, TError> & { queryKey: QueryKey } => {
    const { query: queryOptions } = options || {};

    const queryKey = queryOptions?.queryKey ?? getGetUserUsernameFeedQueryKey(username, params);

    const queryFn: QueryFunction<AsyncReturnType<typeof getUserUsernameFeed>> = () =>
        getUserUsernameFeed(username, params);

    const query = useQuery<AsyncReturnType<typeof getUserUsernameFeed>, TError, TData>(queryKey, queryFn, {
        enabled: !!username,
        ...queryOptions,
    });

    return {
        queryKey,
        ...query,
    };
};

/**
 * Get an activity by ID
 * @summary Activity by ID
 */
export const getActivityId = (id: string) => {
    return customInstance<GetActivityId200>({ url: `/activity/${id}`, method: 'get' });
};

export const getGetActivityIdQueryKey = (id: string) => [`/activity/${id}`];

export const useGetActivityId = <TData = AsyncReturnType<typeof getActivityId>, TError = ApiErrorResponse>(
    id: string,
    options?: { query?: UseQueryOptions<AsyncReturnType<typeof getActivityId>, TError, TData> },
): UseQueryResult<TData, TError> & { queryKey: QueryKey } => {
    const { query: queryOptions } = options || {};

    const queryKey = queryOptions?.queryKey ?? getGetActivityIdQueryKey(id);

    const queryFn: QueryFunction<AsyncReturnType<typeof getActivityId>> = () => getActivityId(id);

    const query = useQuery<AsyncReturnType<typeof getActivityId>, TError, TData>(queryKey, queryFn, {
        enabled: !!id,
        ...queryOptions,
    });

    return {
        queryKey,
        ...query,
    };
};

/**
 * Get a paginated activities for a user
 * @summary User activity feed
 */
export const getActivity = (params?: GetActivityParams) => {
    return customInstance<GetActivity200>({ url: `/activity`, method: 'get', params });
};

export const getGetActivityQueryKey = (params?: GetActivityParams) => [`/activity`, ...(params ? [params] : [])];

export const useGetActivity = <TData = AsyncReturnType<typeof getActivity>, TError = ApiErrorResponse>(
    params?: GetActivityParams,
    options?: { query?: UseQueryOptions<AsyncReturnType<typeof getActivity>, TError, TData> },
): UseQueryResult<TData, TError> & { queryKey: QueryKey } => {
    const { query: queryOptions } = options || {};

    const queryKey = queryOptions?.queryKey ?? getGetActivityQueryKey(params);

    const queryFn: QueryFunction<AsyncReturnType<typeof getActivity>> = () => getActivity(params);

    const query = useQuery<AsyncReturnType<typeof getActivity>, TError, TData>(queryKey, queryFn, queryOptions);

    return {
        queryKey,
        ...query,
    };
};
