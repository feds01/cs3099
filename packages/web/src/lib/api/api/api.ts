/**
 * Generated by orval v6.6.4 üç∫
 * Do not edit manually.
 * Iamus API
 * This is a REST API for interfacing with Iamus. This API provides endpoints for interacting with user information, submissions, and reviews.
 * OpenAPI spec version: 1.0.0
 */
import type { GetVersion200, GetOpenapi200 } from '.././models';
import { customInstance } from '.././mutator/custom-instance';
import { useQuery, UseQueryOptions, QueryFunction, UseQueryResult, QueryKey } from 'react-query';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
type AsyncReturnType<T extends (...args: any) => Promise<any>> = T extends (...args: any) => Promise<infer R> ? R : any;

/**
 * Get the current version of the service
 * @summary Service version
 */
export const getVersion = () => {
    return customInstance<GetVersion200>({ url: `/version`, method: 'get' });
};

export const getGetVersionQueryKey = () => [`/version`];

export const useGetVersion = <TData = AsyncReturnType<typeof getVersion>, TError = unknown>(options?: {
    query?: UseQueryOptions<AsyncReturnType<typeof getVersion>, TError, TData>;
}): UseQueryResult<TData, TError> & { queryKey: QueryKey } => {
    const { query: queryOptions } = options || {};

    const queryKey = queryOptions?.queryKey ?? getGetVersionQueryKey();

    const queryFn: QueryFunction<AsyncReturnType<typeof getVersion>> = () => getVersion();

    const query = useQuery<AsyncReturnType<typeof getVersion>, TError, TData>(queryKey, queryFn, queryOptions);

    return {
        queryKey,
        ...query,
    };
};

/**
 * Get the current service OpenAPI schema
 * @summary Service OpenAPI schema
 */
export const getOpenapi = () => {
    return customInstance<GetOpenapi200>({ url: `/openapi`, method: 'get' });
};

export const getGetOpenapiQueryKey = () => [`/openapi`];

export const useGetOpenapi = <TData = AsyncReturnType<typeof getOpenapi>, TError = unknown>(options?: {
    query?: UseQueryOptions<AsyncReturnType<typeof getOpenapi>, TError, TData>;
}): UseQueryResult<TData, TError> & { queryKey: QueryKey } => {
    const { query: queryOptions } = options || {};

    const queryKey = queryOptions?.queryKey ?? getGetOpenapiQueryKey();

    const queryFn: QueryFunction<AsyncReturnType<typeof getOpenapi>> = () => getOpenapi();

    const query = useQuery<AsyncReturnType<typeof getOpenapi>, TError, TData>(queryKey, queryFn, queryOptions);

    return {
        queryKey,
        ...query,
    };
};
