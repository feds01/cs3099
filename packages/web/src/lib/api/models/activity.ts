/**
 * Generated by orval v6.6.4 üç∫
 * Do not edit manually.
 * Iamus API
 * This is a REST API for interfacing with Iamus. This API provides endpoints for interacting with user information, submissions, and reviews.
 * OpenAPI spec version: 1.0.0
 */
import type { ActivityKind } from './activityKind';
import type { ActivityReference } from './activityReference';
import type { ActivityType } from './activityType';
import type { User } from './user';

export interface Activity {
    id: string;
    owner: User;
    type: ActivityType;
    kind: ActivityKind;
    createdAt: number;
    updatedAt: number;
    message: string;
    references: ActivityReference[];
}
