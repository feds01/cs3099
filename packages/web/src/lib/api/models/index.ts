export * from './successStatus';
export * from './skipQuery';
export * from './flagSchema';
export * from './publicationActivityType';
export * from './successStatusObject';
export * from './takeQuery';
export * from './activityReference';
export * from './publicationActivity';
export * from './userActivity';
export * from './commentActivity';
export * from './reviewActivity';
export * from './commentActivityType';
export * from './multipleErrorMessage';
export * from './activityType';
export * from './responseError';
export * from './emailValidation';
export * from './userPatchRequest';
export * from './tokenRequest';
export * from './userRole';
export * from './userActivityType';
export * from './userLogin';
export * from './activityKind';
export * from './user';
export * from './userRegistration';
export * from './reviewActivityType';
export * from './followerCount';
export * from './publication';
export * from './uploadResource';
export * from './fileResponseType';
export * from './createPublicationRequest';
export * from './revisePublicationRequest';
export * from './directoryResponseType';
export * from './directoryResponseEntriesItem';
export * from './directoryResponseEntriesItemType';
export * from './fileResponse';
export * from './directoryResponse';
export * from './reviewStatus';
export * from './review';
export * from './comment';
export * from './patchComment';
export * from './sgUser';
export * from './commentAnchor';
export * from './sgPublication';
export * from './notification';
export * from './noContentResponse';
export * from './sgComment';
export * from './activity';
export * from './followingListData';
export * from './apiErrorResponseErrors';
export * from './apiErrorResponse';
export * from './patchPublicationRequest';
export * from './sgReview';
export * from './userAuthResponseResponse';
export * from './followerListResponse';
export * from './apiErrorResponseStatus';
export * from './createPublicationResponseResponse';
export * from './responseErrorMessage';
export * from './tokenVerificationResponseResponse';
export * from './resourceResponseResponse';
export * from './createCommentRequestBody';
export * from './getVersion200';
export * from './usernameValidation';
export * from './tokenResponseResponse';
export * from './resourceResponseResponseEntry';
export * from './getOpenapi200Schema';
export * from './getSgSsoLoginParams';
export * from './postSgResourcesImportParams';
export * from './postAuthEmailvalidity200';
export * from './postAuthSsoParams';
export * from './postAuthSso200';
export * from './postAuthUsernamevalidity200';
export * from './getSgSsoCallbackParams';
export * from './getUser200';
export * from './getUserParams';
export * from './getUserUsernameReviews200';
export * from './publicationMetadataResponseResponse';
export * from './postSgSsoVerifyParams';
export * from './patchUserUsername200';
export * from './getUserUsernameFollow200';
export * from './getUserUsername200';
export * from './getOpenapi200';
export * from './getUserUsernameFeedParams';
export * from './getUserUsernameFollowingParams';
export * from './getUserUsernameFollowersParams';
export * from './publicationExportResponseResponse';
export * from './getPublicationParams';
export * from './getPublicationUsername200';
export * from './followerListData';
export * from './getPublicationUsernameParams';
export * from './deletePublicationUsernameNameParams';
export * from './getUserUsernameFeed200';
export * from './patchUserUsernameRoleBody';
export * from './getPublication200';
export * from './getPublicationUsernameNameParams';
export * from './getPublicationUsernameNameRevisions200';
export * from './getPublicationUsernameNameTreePathParams';
export * from './getPublicationUsernameNameRevisionsParams';
export * from './postPublicationUsernameNameExportParams';
export * from './apiSuccessResponse';
export * from './getPublicationUsernameNameSources200';
export * from './deletePublicationUsernameNameAll200';
export * from './getPublicationUsernameNameReviewsParams';
export * from './postPublicationUsernameNameReview200';
export * from './postPublicationUsernameNameReviewParams';
export * from './getPublicationUsernameNameReviews200';
export * from './getPublicationbyidId200';
export * from './postPublicationbyidIdExportParams';
export * from './singleErrorMessage';
export * from './postPublicationbyidIdReview200';
export * from './postResourceUploadUsername200';
export * from './getPublicationbyidIdSources200';
export * from './getPublicationbyidIdReviewsParams';
export * from './getPublicationUsernameNameSourcesParams';
export * from './putReviewIdComment200';
export * from './getCommentId200';
export * from './getThreadId200';
export * from './getPublicationbyidIdReviews200';
export * from './patchPublicationUsernameNameParams';
export * from './getReview200';
export * from './patchCommentId200';
export * from './getSearchPublicationParams';
export * from './getSearchPublication200';
export * from './getActivityId200';
export * from './getReviewId200';
export * from './getSearchUserParams';
export * from './getActivityParams';
export * from './getNotificationsParams';
export * from './getActivity200';
export * from './getPublicationUsernameName200';
export * from './getReviewParams';
export * from './getNotifications200';
export * from './getSearchUser200';
export * from './getReviewIdComments200';
export * from './getUserUsernameRole200';
export * from './patchUserUsernameRole200';
export * from './getNotificationsId200';
export * from './postUserUsernameFollow200';
