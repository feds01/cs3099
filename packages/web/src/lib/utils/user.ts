/**
 * Function that generates initials from provided first and last names of
 * a user.
 *
 * @param name The full name of the user.
 * @returns The initials, based on the first and last name provided.
 */
export function getUserInitials(name: string): string {
    return name
        .split(' ')
        .map((name) => name.charAt(0))
        .join('');
}
