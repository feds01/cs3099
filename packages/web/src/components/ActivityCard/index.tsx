import { Activity, ActivityKind, ActivityReference, ActivityType } from '../../lib/api/models';
import { Divider, Typography } from '@mui/material';
import Box from '@mui/material/Box';
import React, { ReactElement } from 'react';
import PublicationLink from '../PublicationLink';
import UserLink from '../UserLink';
import UserAvatar from '../UserAvatar';
import { formatDistance } from 'date-fns';
import { MdAutoFixHigh, MdDeleteOutline, MdRateReview } from 'react-icons/md';
import { BsStars } from 'react-icons/bs';
import { BiEditAlt } from 'react-icons/bi';
import { IconBaseProps } from 'react-icons';
import ReviewLink from '../ReviewLink';

interface ActivityCardProps {
    activity: Activity;
    withDivider?: boolean;
}

interface LinkWrapperProps {
    reference: ActivityReference;
}

interface ActivityIconProps {
    kind: ActivityKind;
    type: ActivityType;
    iconProps: IconBaseProps;
}

/** Essentially a lookup table of what icon to use when given the type and kind of an Activity */
function ActivityIcon({ kind, type, iconProps }: ActivityIconProps): ReactElement {
    switch (kind) {
        case 'delete':
            return <MdDeleteOutline {...iconProps} />;
        case 'create':
            switch (type) {
                case 'review':
                    return <MdRateReview {...iconProps} />;
                default:
                    return <BsStars {...iconProps} />;
            }
        case 'update':
            return <MdAutoFixHigh {...iconProps} />;
        case 'revise':
            return <BiEditAlt {...iconProps} />;
    }
}

/** Utility component to help render references to various objects on the platform */
function LinkWrapper({ reference }: LinkWrapperProps): ReactElement {
    switch (reference.type) {
        case 'publication':
            return <PublicationLink publication={reference.document} />;
        case 'user':
            return <UserLink user={reference.document} />;
        case 'review':
            return <ReviewLink {...reference.document} />;
        case 'comment':
            // @@Future: we currently don't have a way to jump to specific comments. we could
            //           do this in a special view on reviews where it jumps to the particular comment
            //           with that id?
            return <>comment</>;
    }
}

function mapReferences(parts: string[], references: ActivityReference[]): ReactElement[] {
    return parts.map((value, index, size) => {
        if (index < size.length - 1) {
            return (
                <React.Fragment key={index}>
                    {value}
                    <LinkWrapper reference={references[index]} />
                </React.Fragment>
            );
        }

        return <React.Fragment key={index}>{value}</React.Fragment>;
    });
}

export default function ActivityCard({
    activity: { message, owner, updatedAt, references, kind, type },
    withDivider = true,
}: ActivityCardProps): ReactElement {
    const parts = message.split(/<\d+>/);

    if (parts.length !== references.length + 1) {
        throw new Error('Invalid number of references in relation to expected');
    }

    return (
        <Box sx={{ m: `8px 0 !important` }}>
            <Box sx={{ display: 'flex', flexDirection: 'row', alignItems: 'flex-start', p: 1 }}>
                <UserAvatar {...owner} displayName={false} size={40} />
                <Box sx={{ width: '100%', paddingLeft: 1 }}>
                    <Box sx={{ display: 'flex', flexDirection: 'row', justifyContent: 'space-between' }}>
                        <span style={{ display: 'flex', flexDirection: 'row' }}>
                            <Typography variant={'body1'}>{owner.name}</Typography>
                            &nbsp;
                            <UserLink user={owner} />
                        </span>
                        <Typography sx={{ color: 'dimgray' }}>
                            {formatDistance(updatedAt, new Date(), { addSuffix: true })}
                        </Typography>
                    </Box>
                    <Box sx={{ display: 'flex', flexDirection: 'row', pt: 1, alignItems: 'center' }}>
                        <ActivityIcon iconProps={{ size: 20, style: { paddingRight: 4 } }} kind={kind} type={type} />
                        <Typography variant={'body1'}>{mapReferences(parts, references)}</Typography>
                    </Box>
                </Box>
            </Box>
            {withDivider && <Divider />}
        </Box>
    );
}
