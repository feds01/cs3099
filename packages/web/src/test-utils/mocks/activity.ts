import { Activity } from '../../lib/api/models';
import { getGetActivityIdMock } from '../../lib/api/activity/activity.msw';

/**
 * Function use to create a mocked activity object. The function also
 * accepts a base DTO (Data Transfer Object) which will be used to
 * override some of the generated fields by the mock so that they
 * can be guaranteed to be the same every mock generation. This function
 * does not generate references
 *
 * @param baseDto - Base Data Transfer Object that's used to override generated fields.
 * @returns a mocked user DTO
 */
export function mockActivity(baseDto?: Partial<Activity>): Activity {
    const { references, ...mockedActivity } = getGetActivityIdMock().activity;

    return { ...mockedActivity, references: [], ...baseDto };
}
