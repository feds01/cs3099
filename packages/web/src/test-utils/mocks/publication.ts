import { Publication } from '../../lib/api/models';
import { getGetPublicationUsernameNameMock } from '../../lib/api/publications/publications.msw';

/**
 * Function use to create a mocked publication object. The function also
 * accepts a base DTO (Data Transfer Object) which will be used to
 * override some of the generated fields by the mock so that they
 * can be guaranteed to be the same every mock generation.
 *
 * @param baseDto - Base Data Transfer Object that's used to override generated fields.
 * @returns a mocked publication DTO
 */
export function mockPublication(baseDto?: Partial<Publication>): Publication {
    const mockedUser = getGetPublicationUsernameNameMock().publication;

    return { ...mockedUser, ...baseDto };
}
