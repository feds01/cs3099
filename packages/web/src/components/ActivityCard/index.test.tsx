import ActivityCard from '.';
import { ActivityKind, ActivityReference, ActivityType } from '../../lib/api/models';
import { mockActivity } from '../../test-utils/mocks/activity';
import { mockPublication } from '../../test-utils/mocks/publication';
import { mockUser } from '../../test-utils/mocks/user';
import renderWithWrapper from '../../test-utils/render';

describe('ActivityCard tests', () => {
    it('renders message with no references', () => {
        const mockedMessage = 'This is an activity';
        const mockedActivity = mockActivity({ message: mockedMessage, references: [] });
        const { getByText } = renderWithWrapper(<ActivityCard activity={mockedActivity} />);

        // Ensure that the message in the file is rendered...
        expect(getByText(mockedMessage)).toBeInTheDocument();
    });

    it('renders message with single reference', () => {
        const mockedUser = mockUser();
        const mockedMessage = '<0> did something';
        const mockedReferences: ActivityReference[] = [
            {
                type: 'user',
                document: mockedUser,
            },
        ];
        const mockedActivity = mockActivity({ message: mockedMessage, references: mockedReferences });

        const { getByText } = renderWithWrapper(<ActivityCard activity={mockedActivity} />);

        // Ensure that the message in the file is rendered...
        expect(getByText(`@${mockedUser.username}`)).toBeInTheDocument();
    });

    it('renders message with multiple references', () => {
        const mockedUser = mockUser();
        const mockedPublication = mockPublication();

        const mockedMessage = '<0> and <1> did something';
        const mockedReferences: ActivityReference[] = [
            {
                type: 'user',
                document: mockedUser,
            },
            {
                type: 'publication',
                document: mockedPublication,
            },
        ];
        const mockedActivity = mockActivity({ message: mockedMessage, references: mockedReferences });

        const { getByText } = renderWithWrapper(<ActivityCard activity={mockedActivity} />);

        // Ensure that the message in the file is rendered...
        expect(getByText(`@${mockedUser.username}`)).toBeInTheDocument();
        expect(getByText(new RegExp(`${mockedPublication.name}`, 'im'))).toBeInTheDocument();
    });

    /**
     * This is a snapshot test example. It will render the component into HTML and
     * then compare to what is saved from the disk
     */
    it('renders message with reference at the end', () => {
        // We have to get a mocked user and then redefine
        const mockedUser = mockUser({ username: 'primary', name: 'Primary', profilePictureUrl: undefined });

        const mockedMessage = 'did something <0>';
        const mockedReferences: ActivityReference[] = [
            {
                type: 'user',
                document: mockedUser,
            },
        ];
        const mockedActivity = mockActivity({
            type: ActivityType.user,
            kind: ActivityKind.create,
            message: mockedMessage,
            owner: mockedUser,
            references: mockedReferences,
        });

        const { asFragment } = renderWithWrapper(<ActivityCard activity={mockedActivity} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
