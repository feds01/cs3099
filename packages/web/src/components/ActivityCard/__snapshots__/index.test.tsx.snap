// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityCard tests renders message with reference at the end 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-inmugw"
  >
    <div
      class="MuiBox-root css-1b8ppav"
    >
      <div
        class="MuiBox-root css-wkpw2c"
      >
        <a
          href="/profile/primary"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1gkyibd-MuiAvatar-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1mv1vdv-MuiTypography-root"
            >
              P
            </p>
          </div>
        </a>
      </div>
      <div
        class="MuiBox-root css-emvk5y"
      >
        <div
          class="MuiBox-root css-9jay18"
        >
          <span
            style="display: flex; flex-direction: row;"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1b7aal1-MuiTypography-root"
            >
              Primary
            </p>
            Â 
            <a
              class="makeStyles-wrapper-12"
              data-mui-internal-clone-element="true"
              href="/profile/primary"
            >
              @primary
            </a>
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1dbd2tr-MuiTypography-root"
          >
            about 52 years ago
          </p>
        </div>
        <div
          class="MuiBox-root css-93748c"
        >
          <svg
            fill="currentColor"
            height="20"
            stroke="currentColor"
            stroke-width="0"
            style="padding-right: 4px;"
            viewBox="0 0 16 16"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.657 6.247c.11-.33.576-.33.686 0l.645 1.937a2.89 2.89 0 0 0 1.829 1.828l1.936.645c.33.11.33.576 0 .686l-1.937.645a2.89 2.89 0 0 0-1.828 1.829l-.645 1.936a.361.361 0 0 1-.686 0l-.645-1.937a2.89 2.89 0 0 0-1.828-1.828l-1.937-.645a.361.361 0 0 1 0-.686l1.937-.645a2.89 2.89 0 0 0 1.828-1.828l.645-1.937zM3.794 1.148a.217.217 0 0 1 .412 0l.387 1.162c.173.518.579.924 1.097 1.097l1.162.387a.217.217 0 0 1 0 .412l-1.162.387A1.734 1.734 0 0 0 4.593 5.69l-.387 1.162a.217.217 0 0 1-.412 0L3.407 5.69A1.734 1.734 0 0 0 2.31 4.593l-1.162-.387a.217.217 0 0 1 0-.412l1.162-.387A1.734 1.734 0 0 0 3.407 2.31l.387-1.162zM10.863.099a.145.145 0 0 1 .274 0l.258.774c.115.346.386.617.732.732l.774.258a.145.145 0 0 1 0 .274l-.774.258a1.156 1.156 0 0 0-.732.732l-.258.774a.145.145 0 0 1-.274 0l-.258-.774a1.156 1.156 0 0 0-.732-.732L9.1 2.137a.145.145 0 0 1 0-.274l.774-.258c.346-.115.617-.386.732-.732L10.863.1z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body1 css-1b7aal1-MuiTypography-root"
          >
            did something 
            <a
              class="makeStyles-wrapper-12"
              data-mui-internal-clone-element="true"
              href="/profile/primary"
            >
              @primary
            </a>
          </p>
        </div>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
    />
  </div>
</DocumentFragment>
`;
